week2/week2.py

```diff
@@ -0,0 +1,26 @@
+class MatchParentheses:
+
+    def __init__(self, base: str, target: str):
+        self.base = base
+        self.target = target
+
+    def match(self) -> int:
```

Comment:
Would be nice if this was a little more descriptive as a method name to describe the output, e.g. get_min_insertions_required_for_match or something